### PR TITLE
Safe remote files and symlink creation

### PR DESCRIPTION
**Pull request by roman-verchikov**
Added classes which handle remote symlink and temporary files creation
which are created on scope enter, and destroyed on scope exit.

### CODE REVIEW

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

File: tests/cloudferrylib/test_file_utils.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        symlink = "_symlink"
+
+        with files.RemoteSymlink(runner, target, symlink):
+            pass
+
+        create_symlink = "ln --symbolic %s %s" % (target, symlink)
+        rm_symlink = "rm -f %s" % symlink
+
+        runner.run.assert_called_once_with(create_symlink)
+        runner.run_ignoring_errors.assert_called_once_with(rm_symlink)
```
Comment (roman-verchikov) [inline tests/cloudferrylib/test_file_utils.py]: Unit tests should be extended to cover `RemoteRunner` and `RemoteTempFile` logic

**Comment by Wasapl** [comment]
Build finished.
Functional test success

**Comment by Wasapl** [comment]
Functional test success

File: cloudferrylib/utils/files.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class RemoteTempFile(object):
+    def __init__(self, runner, filename, text):
+        self.runner = runner
+        self.filename = os.path.join('/tmp', '{}'.format(filename))
+        self.text = text
+
+    def __enter__(self):
+        cmd = "echo '{text}' > {file}".format(text=self.text, file=self.filename)
```
Comment (roman-verchikov) [inline cloudferrylib/utils/files.py]: This will break for binary files

File: cloudferrylib/utils/files.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        self.runner.run(cmd)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.runner.run_ignoring_errors(_unlink(self.symlink))
+        return self
+
+
+class RemoteTempFile(object):
+    def __init__(self, runner, filename, text):
```
Comment (roman-verchikov) [inline cloudferrylib/utils/files.py]: Interface may need to be extended with ability to generate temp name

**Comment by Wasapl** [comment]
Build finished.
Unittest success

**Comment by Wasapl** [comment]
Unittest success

**Comment by Wasapl** [comment]
Build finished.
Functional test success

**Comment by Wasapl** [comment]
Functional test success

**Comment by Wasapl** [comment]
Build finished.
Functional test success

**Comment by Wasapl** [comment]
Functional test success

**Comment by roman-verchikov** [comment]
Does not touch core functionality, can be used as a basis for remote file operations in future. Can be merged.

**Pull request merged**